synapse/models/telco.py

```diff
@@ -59,38 +61,71 @@ def castTelLocal(valu):
 def digits(text):
     return ''.join([c for c in text if c.isdigit()])
 
-class PhoneType(DataType):
-    def norm(self, valu, oldval=None):
-
-        if isinstance(valu, str):
-            valu = int(digits(valu))
-
+def chop_imei(imei):
+    valu = int(imei)
+    tac = int(imei[0:8])
+    snr = int(imei[8:14])
+    cd = int(imei[14:15])
+    return valu, {'subs': {'tac': tac, 'serial': snr, 'cd': cd}}
+
+class Phone(s_types.Type):
```

Comment:
This looks great.  It almost makes me wonder about making more of these sub s_types.Str, but i think in this case it'll be good to keep it separate.  Nicely done.